src/Methods/Update.php

```diff
@@ -220,4 +224,26 @@ private function getRequestBody(?RequestInterface $request)
 
         return json_decode($rawBody, true);
     }
+
+    /** Dispatch Update Event. */
+    private function dispatchUpdateEvent(UpdateObject $update): void
+    {
+        if (! property_exists($this, 'eventEmitter') || ! $this->eventEmitter instanceof EmitterInterface) {
+            return;
+        }
+
+        $eventEmitter = $this->eventEmitter;
+
+        $event = new UpdateEvent($this, $update);
+
+        $eventEmitter->emit($event->cloneWithCustomName(UpdateEvent::NAME));
+        $updateType = $update->objectType();
+        if (is_string($updateType)) {
+            $eventEmitter->emit($event->cloneWithCustomName($updateType));
```

Comment:
Why is the UpdateEvent being reused here? In V4, we dispatch 2-3 different events. Update, Update Type/Sub Message event. We don't use a class, it's simply a string.

Reply:
Hm, makes sense, honestly I thought that league/events v2 supports only class-events without name. I'll update the PR tomorrow 👍

Reply:
Thanks!

Reply:
I just checked docs and I was right: when you emit an event using `league/events v2` it's not possible to specify name and event as separate arguments, only like this:
```php
$emitter->emit(new UpdateEvent());
// or
#mitter->emit('string.event');
```

But it's possible to provide extra arguments: https://event.thephpleague.com/2.0/events/arguments/


I'll try this option. It's sort of a hack, but probably will work for this case.

Reply:
@irazasyed 
your goal is to use the same even names as on SDK v4, right? If so, it already worked (and works), but I've added/improved a test to make sure. We don't even need to use "arguments" feature from league/events v2